apps/homepage/src/app/[lang]/_shared/largeNavbar/largeNavbar.tsx

```diff
@@ -0,0 +1,50 @@
+"use client";
+import styles from "./largeNavbar.module.css";
+import { IconLink } from "@1xdev/ui";
+import buttons from "../header/buttons.json";
+import { UiIconProps } from "@uireact/icons";
+import { usePathname } from "next/navigation";
+import { useMemo } from "react";
+
+type HeaderProps = {
+  loc: {
+    [x in string]: string;
+  };
+};
+
+export function LargeNavbar({ loc }: HeaderProps) {
+  const pathname = usePathname();
+
+  const formattedPath = useMemo(() => {
+    const getFormattedPath = (pathname: string) => {
+      const cleanedPath = pathname.replace(/^\/(es|en)/, "");
+      return cleanedPath === "" ? "/" : cleanedPath;
+    };
+    return getFormattedPath(pathname);
+  }, [pathname]);
```

Comment:
What about just doing inside the `useMemo`...

```
  const cleanedPath = pathname.replace(/^\/(es|en)/, "");
   return cleanedPath === "" ? "/" : cleanedPath;
```

Or better you could move this logic to another file and have it shared between `largeNavbar` and `smallNavbar`.